br/pkg/lightning/restore/precheck.go

```diff
@@ -73,11 +72,11 @@ func NewPrecheckItemBuilderFromConfig(ctx context.Context, cfg *config.Config, o
 	}
 	mdl, err := mydump.NewMyDumpLoader(ctx, cfg, builderCfg.MDLoaderSetupOptions...)
 	if err != nil {
-		if errors.ErrorEqual(err, common.ErrTooManySourceFiles) {
-			gerr = err
-		} else {
+		if mdl == nil {
 			return nil, errors.Trace(err)
 		}
+		// here means the partial result is returned, so we can continue on processing
```

Comment:
no need to check `ReturnPartialResultOnError`? maybe comment why we always use partial result here.

Reply:
Because originally, `NewMyDumpLoader` will always return nil when error occurs.  Now this behavior is changed only the user adds the `ReturnPartialResultOnError` option when constructing the MDL.  So, if returned mdl is not nil, we can infer that users sets the `ReturnPartialResultOnError` option or equivalents.